File: fleman-backend/src/main/java/com/fleman/service/BookingService.java
package com.fleman.service;

import com.fleman.dto.BookingResponseDTO;
import com.fleman.dto.CreateBookingRequest;
import com.fleman.dto.ModifyBookingRequest;

import java.util.List;

public interface BookingService {
    BookingResponseDTO createBooking(CreateBookingRequest request);
    BookingResponseDTO getBookingByConfirmation(String confirmationNo);
    BookingResponseDTO getLastBookingForCustomer(Long customerId);
    List<BookingResponseDTO> getBookingsForCustomer(Long customerId);
    BookingResponseDTO modifyBooking(String confirmationNo, ModifyBookingRequest request);
    BookingResponseDTO cancelBooking(String confirmationNo);
}

File: frontend/src/api/bookingService.js
import { api } from './client';

// POST /api/bookings
// payload: { customerId, carId, pickupHubId, dropHubId, pickupDatetime,
//            returnDatetime, addons: [{ addonId, quantity }], estimatedAmount, remarks }
//
// Note: estimatedAmount is sent for display continuity only — the server
// recomputes rental + add-on totals itself from current rates and never
// trusts a client-submitted price.
export async function createBooking(payload) {
  return api.post('/bookings', payload);
}

// GET /api/bookings/:confirmationNo
export async function getBookingByConfirmation(confirmationNo) {
  return api.get(`/bookings/confirmation/${confirmationNo}`);
}

// GET /api/bookings/me?customerId=
export async function getLastBookingForCustomer(customerId) {
  return api.get('/bookings/me', { params: { customerId } });
}

// GET /api/bookings?customerId= — every booking for the customer, not just
// the last one. Backs the "modify booking" page's picker: a logged-in
// customer sees all their bookings and chooses which to change.
export async function getBookingsForCustomer(customerId) {
  return api.get('/bookings', { params: { customerId } });
}

// PUT /api/bookings/:confirmationNo
export async function modifyBooking(confirmationNo, patch) {
  return api.put(`/bookings/confirmation/${confirmationNo}`, patch);
}

// DELETE /api/bookings/:confirmationNo
export async function cancelBooking(confirmationNo) {
  return api.delete(`/bookings/confirmation/${confirmationNo}`);
}
